lnrpc/neutrinorpc/neutrino.proto

```diff
@@ -0,0 +1,214 @@
+    Status returns the status of the light client neutrino instance,
+    along with height and hash of the best block, and a list of connected
+    peers.
+    */
+    rpc Status (StatusRequest) returns (StatusResponse);
+
+    /*
+    AddPeer adds a new peer that has already been connected to the server.
+    */
+    rpc AddPeer (AddPeerRequest) returns (AddPeerResponse);
+
+    /*
+    DisconnectPeer disconnects a peer by target address. Both outbound and
+    inbound nodes will be searched for the target node. An error message will
+    be returned if the peer was not found.
+    */
+    rpc DisconnectPeer (DisconnectPeerRequest) returns (DisconnectPeerResponse);
+
+    /*
+    IsBanned returns true if the peer is banned, otherwise false.
+    */
+    rpc IsBanned (IsBannedRequest) returns (IsBannedResponse);
+
+    /*
+    GetBlockHeader returns a block header with a particular block hash.
+    */
+    rpc GetBlockHeader (GetBlockHeaderRequest) returns (GetBlockHeaderResponse);
+
+    /*
+    GetBlock returns a block with a particular block hash.
+    */
+    rpc GetBlock (GetBlockRequest) returns (GetBlockResponse);
+
+    /*
+    GetCFilter returns a compact filter from a block.
+    */
+    rpc GetCFilter (GetCFilterRequest) returns (GetCFilterResponse);
+}
+
+message StatusRequest {
+}
+
+message StatusResponse {
+    // Indicates whether the neutrino is active or not.
+    bool active = 1;
+
+    // Is fully synced.
+    bool synced = 2;
+
+    // Best block height.
+    int32 block_height = 3;
+
+    // Best block hash.
+    string block_hash = 4;
+
+    // Connected peers.
+    repeated string peers = 5;
+}
+
+message AddPeerRequest {
+    // Peer to add.
+    string peer_addrs = 1;
+}
+
+message AddPeerResponse {
+}
+
+message DisconnectPeerRequest {
+    // Peer to disconnect.
+    string peer_addrs = 1;
+}
+
+message DisconnectPeerResponse {
+}
+
+message IsBannedRequest {
+    // Peer to lookup.
+    string peer_addrs = 1;
+}
+
+message IsBannedResponse {
+    bool banned = 1;
+}
+
+message GetBlockHeaderRequest {
+    // Block hash in hex notation.
+    string hash = 1;
+}
+
+message GetBlockHeaderResponse {
+    // The block hash (same as provided).
+    string hash = 1;
+
+    // The number of confirmations.
+    int64 confirmations = 2;
+
+    // The block size excluding witness data.
+    int64 strippedsize = 3;
+
+    // The block size (bytes).
+    int64 size = 4;
+
+    // The block weight as defined in BIP 141.
+    int64 weight = 5;
+
+    // The block height or index.
+    int32 height = 6;
+
+    // The block version.
+    int32 version = 7;
+
+    // The block version.
+    string version_hex = 8;
+
+    // The merkle root.
+    string merkleroot = 9;
+
+    // The block time in seconds since epoch (Jan 1 1970 GMT).
+    int64 time = 10;
+
+    // The nonce.
+    uint32 nonce = 11;
+
+    // The bits in hex notation.
+    string bits = 12;
+
+    // The number of transactions in the block.
+    int32 ntx = 13;
+
```

Comment:
I wonder if we should just provide the raw block/header bytes instead? Even if we keep the rest of the fields, we should add the raw hex imo so clients can get all the transaction details, etc.